### PR TITLE
Miguelhar.mod migration

### PR DESCRIPTION
**Pull request by miguelhar**
- Updates the example terraform configuration to integrate missing components like flyte.
- Fixes provisioning the costs buckets part of the storage module even when costs is disabled

### CODE REVIEW

File: bin/module-migration/README.md
Comment (ddl-dclegg) [inline bin/module-migration/README.md:1]: Do we need this?

File: examples/deploy/terraform/cluster/main.tf [hunk 1]
```diff
@@ -83,6 +83,19 @@ module "external_deployments_operator" {
   external_deployments = var.external_deployments_operator
 }
 
+module "flyte" {
+  count                     = var.flyte.enabled ? 1 : 0
+  source                    = "./../../../../modules/flyte"
```
Comment (ddl-dclegg) [inline examples/deploy/terraform/cluster/main.tf:88]: This feels fragile, is there any way we can use [path expressions](https://developer.hashicorp.com/terraform/language/expressions/references#filesystem-and-workspace-info) instead?

**Review by ddl-dclegg — APPROVED** [review]
Looks good, but I have a couple of questions.

File: bin/module-migration/README.md
Comment (miguelhar) [reply bin/module-migration/README.md:1]: yes.

File: examples/deploy/terraform/cluster/main.tf (hunk 1 above)
Comment (miguelhar) [reply examples/deploy/terraform/cluster/main.tf:88]: there is a script that handles the module source. https://github.com/dominodatalab/terraform-aws-eks/blob/main/examples/deploy/set-mod-version.sh.

File: bin/module-migration/README.md
Comment (ddl-dclegg) [reply bin/module-migration/README.md:1]: The added content makes all the difference. Thanks!

**Review by ddl-dclegg — APPROVED** [review]

File: bin/module-migration/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Module Migration Script for terraform-aws-eks
+
+This(`migrate-mods.sh`) script automates the process of updating module versions for infrastructure and EKS clusters. It handles backups, updates to tfvars, and necessary imports and deprecations. It was tested migrating from version `v3.0.5` to `v3.0.22`.
```
[the remark this answers is not in the document]
Comment (miguelhar) [reply bin/module-migration/README.md:3]: yes, thats a typo. in response to the initial question I would say it  can be used to migrate to any versions in between but I did not validate that.

File: bin/module-migration/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+4. Move to the deployment directory and initialize components.
+```shell
+cd "$DEPLOY_DIR"
+./tf.sh all init
+```
+5. Plan, **Review Plan** and Apply each component.
+```shell
+./tf.sh infra plan
+##REVIEW INFRA PLAN
+# Whats expected
```
[the remark this answers is not in the document]
Comment (miguelhar) [reply bin/module-migration/README.md:60]: Added documentation and conditionals to address this

**Comment by ddl-jacob** [comment]
looks good to me, :shipit:

**Review by ddl-jacob — APPROVED** [review]

File: bin/module-update/module-update.sh [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  declare -A tfvars_paths=(
+    ["$INFRA_DIR"]=$INFRA_VARS
+    ["$CLUSTER_DIR"]="$CLUSTER_VARS"
+    ["$NODES_DIR"]="$NODES_VARS"
+  )
+
+  for mod_dir in "${!tfvars_paths[@]}"; do
+    vars_path="${tfvars_paths[$mod_dir]}"
+    if [ "$(basename $mod_dir)" = "cluster" ] && is_vpc_cni_installed; then
+      echo
```
Comment (Secretions) [inline bin/module-update/module-update.sh:145]: Is this a placeholder for adding to the addon list or something?

Comment (miguelhar) [reply bin/module-update/module-update.sh:145]: Initially I thought to make it smart but it is not trivial to edit the hcl nested value. I want to just print a message but forgot to finish the line.

File: bin/module-update/module-update.sh [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  terraform -chdir="$UPGRADE_DIR" init -backend=false -from-module="github.com/dominodatalab/terraform-aws-eks.git//examples/deploy?ref=${MOD_VERSION}"
+
+  printf "Updating module source...\n"
+  update_mod_source "$UPGRADE_DIR" "$MOD_VERSION"
+}
+
+backup_files() {
+  if [ ! -d "$BACKUP_DIR" ] || [ -z "$(ls -A "$BACKUP_DIR")" ]; then
+    printf "Backing up %s to %s.\n" "$DEPLOY_DIR" "$BACKUP_DIR"
+    cp -r "$DEPLOY_DIR" "$BACKUP_DIR"
```
Comment (Secretions) [inline bin/module-update/module-update.sh:114]: Would be good to do trailing slash here so you don't end up with `$BACKUP_DIR/$DEPLOY_DIR` if this gets retried.

**Review by Secretions — APPROVED** [review]

**Pull request merged**